src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs

```diff
@@ -0,0 +1,238 @@
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.ValueTracking
+{
+    [Export(typeof(ICommandHandler))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [Name(PredefinedCommandHandlerNames.ShowValueTracking)]
+    internal class ValueTrackingCommandHandler : ICommandHandler<ValueTrackingEditorCommandArgs>
+    {
+        private readonly IAsyncServiceProvider _serviceProvider;
+        private readonly SVsServiceProvider _serviceProvider1;
+        private readonly IThreadingContext _threadingContext;
+        private readonly ClassificationTypeMap _typeMap;
+        private readonly IClassificationFormatMapService _classificationFormatMapService;
+        private readonly IGlyphService _glyphService;
+        private RoslynPackage? _roslynPackage;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ValueTrackingCommandHandler(
+            SVsServiceProvider serviceProvider,
+            IThreadingContext threadingContext,
+            ClassificationTypeMap typeMap,
+            IClassificationFormatMapService classificationFormatMapService,
+            IGlyphService glyphService)
+        {
+            _serviceProvider = (IAsyncServiceProvider)serviceProvider;
+            _serviceProvider1 = serviceProvider;
+            _threadingContext = threadingContext;
+            _typeMap = typeMap;
+            _classificationFormatMapService = classificationFormatMapService;
+            _glyphService = glyphService;
+        }
+
+        public string DisplayName => "Go to value tracking";
+
+        public CommandState GetCommandState(ValueTrackingEditorCommandArgs args)
+            => CommandState.Available;
+
+        public bool ExecuteCommand(ValueTrackingEditorCommandArgs args, CommandExecutionContext executionContext)
+        {
+            var cancellationToken = executionContext.OperationContext.UserCancellationToken;
+            var caretPosition = args.TextView.GetCaretPoint(args.SubjectBuffer);
+            if (!caretPosition.HasValue)
+            {
+                return false;
+            }
+
+            var textSpan = new TextSpan(caretPosition.Value.Position, 0);
+            var sourceTextContainer = args.SubjectBuffer.AsTextContainer();
+            var document = sourceTextContainer.GetOpenDocumentInCurrentContext();
+            if (document is null)
+            {
+                return false;
+            }
+
+            var selectedSymbol = _threadingContext.JoinableTaskFactory.Run(() => GetSelectedSymbolAsync(textSpan, document, cancellationToken));
```

Comment:
Have we talked to the platform folks about it being possible to have an async command handler so we don't need to use JTF for this?

Reply:
Moved this to just RunAsync, similar to https://github.com/madskristensen/AsyncToolWindowSample/blob/master/src/Commands/ShowToolWindow.cs

Reply:
Thanks, I like this a lot better. It being fire-and-forget seems fine compared to blocking the caller on async work.